PFR/Mathlib/Probability/Independence/Basic.lean

```diff
@@ -227,133 +227,89 @@ variable {ι Ω : Type*} {κ : ι → Type*} {α : ∀ i, κ i → Type*} [Measu
   [IsProbabilityMeasure μ] {m : ∀ i j, MeasurableSpace (α i j)} {f : ∀ i j, Ω → α i j}
   [Fintype ι] [∀ i, Fintype (κ i)]
 
--- Below, two approaches
-
-/-- If a family of functions `i j ↦ f i j` is independent, then the family of functions
-`i ↦ ∀ j, f i j` is independent. -/
-lemma iIndepFun.pi
-    (hf : iIndepFun (fun ij : Σ i, κ i ↦ m ij.1 ij.2) (fun ij : Σ i, κ i ↦ f ij.1 ij.2) μ) :
-    iIndepFun (fun i ↦ MeasurableSpace.pi) (fun i ω j ↦ f i j ω) μ := by
-  rw [iIndepFun_iff_measure_inter_preimage_eq_mul] at hf ⊢
-  rintro s t ht
-  -- Here, we want to WLOG reduce to the case where the `t i` are boxes of the form
-  -- `Set.univ.pi (fun j : κ i → (u i j : Set (α i j)))`
-  sorry
-
-lemma measurable_sigmaCurry :
-    Measurable (Sigma.curry : (∀ ij : Σ i, κ i, α ij.1 ij.2) → ∀ i j, α i j) := sorry
-
-lemma measurable_sigmaUncurry :
-    Measurable (Sigma.uncurry : (∀ i j, α i j) → (∀ ij : Σ i, κ i, α ij.1 ij.2)) := sorry
-
-/-- If a family of functions `i j ↦ f i j` is independent, then the family of functions
-`i ↦ ∀ j, f i j` is independent. -/
-lemma iIndepFun.pi'
-    (hf : iIndepFun (fun ij : Σ i, κ i ↦ m ij.1 ij.2) (fun ij : Σ i, κ i ↦ f ij.1 ij.2) μ) :
-    iIndepFun (fun i ↦ MeasurableSpace.pi) (fun i ω j ↦ f i j ω) μ := by
-  rw [iIndepFun_iff_pi_map_eq_map] at hf ⊢
-  symm
-  calc
-    μ.map (fun ω i j ↦ f i j ω)
-      = (μ.map fun ω (ij : Σ i, κ i) ↦ f ij.1 ij.2 ω).map Sigma.curry := by
-        rw [Measure.map_map]; rfl
-        exact measurable_sigmaCurry
-        sorry
-    _ = _ := by rw [←hf]
-    _ = _ := ?_
-  sorry
-  sorry
-  sorry
```

Comment:
I think we should not delete these. `iIndepFun.pi` is the correct lemma to prove, even if it's not the one we've managed to prove first.

Reply:
unstaged this file now that #131 exists